contracts/src/v0.8/feeds/DualAggregator.sol

```diff
@@ -816,25 +816,24 @@ contract DualAggregator is OCR2Abstract, OwnerIsCreator, AggregatorV2V3Interface
         _payTransmitter(s_hotVars, report_.juelsPerFeeCoin, uint32(initialGas), msg.sender);
         return;
       }
-      // In case the report doesn't exist, lock the primary feed
-      s_primaryLocked = true;
-    } else {
-      // In case the sender is the primary proxy, unlock the latest report
-      s_primaryLocked = false;
     }
-
     // Report epoch and round
     uint40 epochAndRound = uint40(uint256(reportContext[1]));
 
-    if (epochAndRound < s_hotVars.latestEpochAndRound) {
-      revert StaleReport();
-    }
+    if (epochAndRound != s_hotVars.latestEpochAndRound || !s_primaryLocked) {
+      if (epochAndRound <= s_hotVars.latestEpochAndRound) {
+        revert StaleReport();
+      }
 
-    // Verify signatures attached to report
-    _verifySignatures(reportContext, report, rs, ss, rawVs);
+      // Verify signatures attached to report
+      _verifySignatures(reportContext, report, rs, ss, rawVs);
 
-    int192 juelsPerFeeCoin = _report(s_hotVars, reportContext[0], epochAndRound, report_, isSecondary);
-    _payTransmitter(s_hotVars, juelsPerFeeCoin, uint32(initialGas), msg.sender);
+      _report(s_hotVars, reportContext[0], epochAndRound, report_, isSecondary);
+    }
+
+    // Lock/unlock the primary feed and pay the transmitters
+    s_primaryLocked = isSecondary;
```

Comment:
This is not easy to understand, as we're only locking for a single block. Let's change this variable to something more direct, saying that the last update was from the secondary feed, and then communicate the implications of this on the reading side

Reply:
That's true, renamed it to `s_latestSecondary`